File: src/com/suushiemaniac/lang/json/value/JsonNull.java
package com.suushiemaniac.lang.json.value;

import com.suushiemaniac.lang.json.JSON;

public class JsonNull extends JsonType {
    public JsonNull() {
        this(null);
    }

    public JsonNull(JSON parent) {
        super(parent);
    }

    @Override
    public Object nullValue() {
        return null;
    }

	@Override
	public Object toNative() {
		return this.nullValue();
	}

	@Override
    public String stringValue() {
        return "null";
    }

	@Override
	public float floatValue() {
		return 0f;
	}

	@Override
	public int intValue() {
		return 0;
	}

	@Override
	public boolean booleanValue() {
		return false;
	}

	public String toString() {
        return "null";
    }

    @Override
    public boolean equals(Object other) {
        return other instanceof JsonNull || other == null;
    }
}

File: src/main/kotlin/com/suushiemaniac/lang/json/value/JsonNull.kt
package com.suushiemaniac.lang.json.value

import com.suushiemaniac.lang.json.JSON

class JsonNull private constructor(parent: JSON? = null) : JsonType(parent) {
    override fun nullValue(): Any {
        return Unit
    }

    override fun toNative(): Any {
        return this.nullValue()
    }

    override fun stringValue(): String {
        return "null"
    }

    override fun floatValue(): Float {
        return 0f
    }

    override fun intValue(): Int {
        return 0
    }

    override fun booleanValue(): Boolean {
        return false
    }

    override fun toString(): String {
        return "null"
    }

    override fun equals(other: Any?): Boolean {
        return other is JsonNull || other == null
    }

    override fun hashCode(): Int {
        return 0
    }

    companion object {
        val INST = JsonNull()
    }
}
